src/vec_bcd_ppc.h

```diff
@@ -461,6 +556,244 @@ vec_cbcdaddcsq (vBCD_t *cout, vBCD_t a, vBCD_t b)
+ * The net seems to be that for BCD multiple precision difference to
+ * work correctly, the larger magnitude must be the first
+ * operand.
+ *
+ * \subsubsection bcd128_extended_0_2_1 Vector BCD Multiply Quadword example
+ *
+ * TBD
+ * BCD multiply and divide operations are not directly supported
+ * in the current PowerISA. Decimal multiply and divide are
+ * supported in the Decimal Floating-point (DFP) Unit (DFU), as well
+ * as conversion to and from signed (unsigned) BCD.
+ *
+ * So BCD multiply and divide operations can be routed through the
+ * DFU with a few caveats.
+ * - DFP Extended format supports up to 34 digits precision
+ * - DFP significand represent digits to the <I>left</I> of the
+ * implied decimal point.
+ * - DFP finite number are not normalize.
+ *
+ * This allows DFP to represent decimal integer and fixed point
+ * decimal values with a preferred exponent of 0.
+ * The DFU will maintain this preferred exponent for DPF arithmetic
+ * operations until:
+ * - An arithmetic operation involves a operand with a non-zero
+ * exponent.
+ * - A divide operation generates a result with fractional digits
+ * - A multiply operation generates a result that exceeds 34 digits.
+ *
+ * The implementation can insure that input operands are derived from
+ * 31-digit BCD values. The results of any divide operations can be
+ * truncated back to decimal integer with the preferred 0 exponent.
+ * This can be achieved with the DFP Quantize Immediate instruction,
+ * specifying the ideal exponent of 0 and a rounding mode of
+ * <I>round toward 0</I> (see vec_quantize0_Decimal128()).
+ * This allows the following implementation:
+ * \code
+static inline vBCD_t
+vec_bcddiv (vBCD_t a, vBCD_t b)
+{
+  vBCD_t t;
+  _Decimal128 d_t, d_a, d_b;
+  d_a = vec_BCD2DFP (a);
+  d_b = vec_BCD2DFP (b);
+  d_t = vec_quantize0_Decimal128 (d_a / d_b);
+  t = vec_DFP2BCD (d_t);
+  return (t);
+}
+ * \endcode
+ *
+ * The multiply case is bit more complicated as we need to produce up
+ * to 62 digit results without losing precision and DFP only supports
+ * 34 digits. This requires splitting the input operands into groups
+ * of digits where partial products of any combination of these groups
+ * is guaranteed not exceed 34 digits.
+ *
+ * One way to do this is split each 31-digit operand into two 16-digit
+ * chunks (actually 15 and 16-digits). These chunks are converted to
+ * DFP extended format and multiplied to product four 32-digit partial
+ * products. These partial products can be aligned and summed to
+ * produce the high  and low 31-digits of the full 62-digit product.
+ * This is the basis for vec_bcd_mul(), vec_bcdmulh(), and
+ * vec_cbcdmul().
+ *
+ * A simple vec_and() can be used to isolate the low order 16 BCD
+ * digits. It is simple at this point to detect if both operands are
+ * 16-digits or less by comparing the original operand to isolate
+ * value. In this case the product can not exceed 32 digits and we
```

Comment:
s/to isolate value/to the isolated value/ ?

Reply:
OK